sa/storage-authority.go

```diff
@@ -213,6 +213,36 @@ func (ssa *SQLStorageAuthority) GetAuthorization(id string) (authz core.Authoriz
 	return
 }
 
+// GetAuthorizationsByDomain obtains all authorizations for a domain name
+func (ssa *SQLStorageAuthority) GetAuthorizationsByDomain(domain core.AcmeIdentifier) ([]core.Authorization, error) {
```

Comment:
Sorry, I just looked deeper. 

This could return a huge amount of authorizations, no? Are we sure this is what we want? We'd have to paginate these, at the very least.

Are we really sure we want all authorizations? Could we return only ones that are not too old to be used?

I'm starting to be convinced that there should be just a RevokeAuthorizationsForDomain call.

@jsha, what do you think?